### PR TITLE
Dynamic item count revisited

### LINKED ISSUE #97: Dynamic item count
**Current behavior**
Currently the hook initializes with a given number of items, however it is not possible to increase or decrease those items to suit a dynamic list.

**Desired behavior**
The hook should be able to handle lists that change in size.

### PR DESCRIPTION
**Pull request by corymharper**
Currently the hook can only handle a list that does not change size, this PR fixes that.

Closes #97  
Closes #78

### CODE REVIEW

**Review by WesCossick — CHANGES_REQUESTED** [review]
The `yarn.lock` files shouldn't need to be updated in this PR. Also; it looks like you've organized the tests by grouping them into `describe()` blocks. Can you do this in a separate PR? It makes it hard to see what is actually being changed to fix the dynamic item bug.

**Comment by corymharper** [comment]
I'm not certain why it thinks I have changed the lock file, I copied it straight from master to this branch…

**Review by WesCossick — CHANGES_REQUESTED** [review]
The `yarn.lock` file still has changes; it should not have any.

**Comment by corymharper** [comment]
> The `yarn.lock` file still has changes; it should not have any.

My `yarn.lock` is identical to the `master` branch `yarn.lock`, I'm not certain why it thinks there are changes. The versions it is listing in the diff are not the versions on master branch currently.

**Comment by WesCossick** [comment]
You may need to merge `master` into your branch.

**Comment by corymharper** [comment]
> You may need to merge `master` into your branch.

I ended up rebasing off the most current HEAD of master

File: test/use-dropdown-menu.test.tsx [hunk 1]
```diff
@@ -195,3 +195,32 @@ it('closes the menu after clicking the button when the menu is open', () => {
...
+	const addItemButton = component.find('#add-item');
+
+	addItemButton.simulate('click');
+
+	const fourthMenuItem = component.find('#menu-item-4');
+
+	expect(fourthMenuItem.prop('role')).toBe('menuitem');
+});
+
+it('can navigate to a new item', () => {
```
Comment (WesCossick) [inline test/use-dropdown-menu.test.tsx]:
```suggestion
it('can navigate to a dynamically-added item', () => {
```

**Review by WesCossick — CHANGES_REQUESTED** [review]

**Pull request merged**